src/sinks/aws_cloudwatch_logs/retry.rs

```diff
@@ -86,13 +87,13 @@ impl<T: Send + Sync + 'static> RetryLogic for CloudwatchRetryLogic<T> {
                 }
                 SdkError::ConstructionFailure(_) => false,
             },
-            CloudwatchError::CreateStream(err) => match err {
+            CloudwatchError::CreateStream(sdk_err) => match sdk_err {
                 SdkError::ServiceError { err, raw: _ } => match err.kind {
                     CreateLogStreamErrorKind::ServiceUnavailableException(_) => {
                         error!(message = "Create stream service unavailable.", %error);
                         true
                     }
-                    _ => false,
+                    _ => is_retriable_error(sdk_err),
                 },
                 SdkError::TimeoutError(_) => true,
```

Comment:
The below `TimeoutError`, `DispatchError`, `ResponseError`, etc. seem to be handled by `is_retriable_error`. Would it make sense to just delegate those ones as well? 

Seemingly even `ServiceError` is already handled by `retriable_error` too 🤔 

Should this just delegate wholesale like the `aws_sqs` source? There doesn't seem to be any service specific error handling here.

Reply:
It still has some service specific stuff (anything using the error under `ServiceError` is service specific), but I can clean this up a bit more since a lot is duplicated